administration/src/components/generation/GenerationController.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {useState} from "react";
+import {Spinner} from "@blueprintjs/core";
+import {CardType} from "../../models/CardType";
+import {CardCreationModel} from "./CardCreationModel";
+import {add} from 'date-fns';
+import GenerationForm from "./GenerationForm";
+import {useApolloClient} from "@apollo/client";
+import {AppToaster} from "../AppToaster";
+import GenerationFinished from "./GenerationFinished";
+import downloadDataUri from "./downloadDataUri";
+import generateCards from "./generateCards";
+
+let idCounter = 0;
+
+const createEmptyCard = (): CardCreationModel => ({
+    id: idCounter++,
+    forename: "",
+    surname: "",
+    expirationDate: add(Date.now(), {years: 2}),
+    cardType: CardType.standard
+});
+
+enum Mode {
+    input,
+    loading,
+    finished
+}
+
+const GenerationController = () => {
+    const [cardCreationModels, setCardCreationModels] = useState([createEmptyCard()]);
+    const client = useApolloClient();
+    const [mode, setMode] = useState(Mode.input);
+    const confirm = async () => {
+        try {
+            setMode(Mode.loading)
+            const pdfDataUri = await generateCards(client, cardCreationModels)
+            downloadDataUri(pdfDataUri, "ehrenamtskarten.pdf")
+            setMode(Mode.finished)
+        } catch (e) {
+            console.error(e)
+            AppToaster.show({message: "Something went wrong.", intent: "danger"})
```

Comment:
I think the rest of our UI is in German, right?

```suggestion
            AppToaster.show({message: "Etwas ist schiefgegangen.", intent: "danger"})
```